libavfilter/dnn/dnn_backend_common.c

```diff
@@ -69,3 +69,66 @@ DNNReturnType ff_dnn_fill_task(TaskItem *task, DNNExecBaseParams *exec_params, v
 
     return DNN_SUCCESS;
 }
+
+static void *async_thread_routine(void *args)
+{
+    DNNAsyncExecModule *async_module = args;
+    void *request = async_module->args;
+
+    async_module->start_inference(request);
+    async_module->callback(request);
+#if HAVE_PTHREAD_CANCEL
+    pthread_exit(0);
+#endif
+}
+
+DNNReturnType ff_init_async_attributes(DNNAsyncExecModule *async_module)
+{
+    if (!async_module) {
+        return DNN_ERROR;
+    }
+#if HAVE_PTHREAD_CANCEL
+    pthread_attr_init(&async_module->thread_attr);
+    pthread_attr_setdetachstate(&async_module->thread_attr, PTHREAD_CREATE_DETACHED);
```

Comment:
not sure if the two functions are supported on Windows.

Reply:
I guess no, but i can try to push this commit first to double check it.

if no, we might consider the default thread.

Reply:
@xuguangxin do we now have windows build in CI?  I heard that we have enabled the windows build, if so, could you please try https://github.com/intel-media-ci/ffmpeg/pull/430, thanks.

Reply:
No, not yet. Windows CI is harder than we expected. If you just want check it's break windows or not. you can ask fan1x.li@intel.com for help.

Reply:
checked with Fan, HAVE_PTHREAD_CANCEL is not enabled in Windows, what enabled is PTHREAD_CANCEL_ENABLE.

@shubhanshu02  to make the async support available for future extending in Windows (we can define HAVE_PTHREAD_CANCEL for windows whenever we want to support async), we'd better do not use the functionalities not defined in https://github.com/FFmpeg/FFmpeg/blob/master/compat/w32pthreads.h 

in other words, we could not use PTHREAD_CREATE_DETACHED, we may do phtread_join at the beginning of ff_dnn_execute_model_async_tf or get_async functions (actually, you may find a better place in your code) .

Reply:
Sure, I'll correct this part.

I think joining the thread in the function `ff_dnn_start_inference_async` so as to keep the threading logic separate in the `dnn_backend_common`. 

One question, do we have to check PTHREAD_CANCEL_ENABLE in addition to HAVE_PTHREAD_CANCEL in the code where we use pthread functions?

Reply:
sure, it is a better place, just to make sure the last thread is joined.

we now just check HAVE_PTHREAD_CANCEL , to make the thing simple.

Reply:
Okay, thanks. I'll update the patches and let you know.

Reply:
Corrected.